toucan_connectors/one_drive/one_drive_connector.py

```diff
@@ -0,0 +1,103 @@
+import os
+from pathlib import Path
+from typing import Optional
+
+import pandas as pd
+import requests
+from pydantic import Field
+
+from toucan_connectors.oauth2_connector.oauth2connector import (
+    OAuth2Connector,
+    OAuth2ConnectorConfig,
+)
+from toucan_connectors.toucan_connector import (
+    ConnectorSecretsForm,
+    ToucanConnector,
+    ToucanDataSource,
+)
+
+API_BASE_ROUTE: str = 'https://graph.microsoft.com/v1.0/me'
+AUTHORIZATION_URL: str = 'https://login.microsoftonline.com/common/oauth2/v2.0/authorize'
+TOKEN_URL: str = 'https://login.microsoftonline.com/common/oauth2/v2.0/token'
+SCOPE: str = 'offline_access Files.Read'
```

Comment:
Perhaps we can make this field configurable. [https://docs.microsoft.com/en-us/onedrive/developer/rest-api/getting-started/msa-oauth?view=odsp-graph-online](Here) I can see that the scope can be restricted to specific folders. 

I suggest we add it as a field in the connector 
`
scope: str = Field(        None, Title='Scope', description='The scope the integration', placeholder='refresh_token'    )`

Reply:
Good idea but I think the mimimum placeholder should be `offline_access` (that means refresh_token) + `onedrive.readonly` (I think it will help conceptors who are not familliar with Oauth).

Ok for you ?

Reply:
Sounds good.